src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -205,7 +203,15 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
           (editors ? ViewportOp(sid, vid, Viewport.Watch)).mapTo[Result].map {
             case ok: Ok with Viewport.Events =>
               ok.stream
-                .map(u => ViewportEvent(u.id, serial = u.change.map(_.id)))
+                .map(u =>
+                  ViewportEvent(
+                    u.id,
+                    serial = u.change.map(_.id),
+                    offset = u.change.map(_.offset),
+                    length = u.change.map(_.length),
```

Comment:
Actually, I think this is wrong.  These needs to be the viewport offset and length, not the change offset and length.  Reason for this is that the viewport offset can change if the viewport is floating.